src/rpc/mining.cpp

```diff
@@ -1289,7 +1291,7 @@ static RPCHelpMan getnewblockhex()
                             },
                         },
                         "proposed_parameters"},
-                        {"commit_data", RPCArg::Type::STR_HEX, RPCArg::Optional::OMITTED_NAMED_ARG, "Data in hex to be committed to in an additional coinbase output."},
+                    {"commit_data", RPCArg::Type::STR_HEX, RPCArg::Optional::OMITTED_NAMED_ARG, "Array of additional hex data to be committed to in additional coinbase outputs. Separate each hex commitment with one or more spaces."}
```

Comment:
Not sure I like the implicit "multiple pushes" when you push mutliple hexes with spaces.

If I would pass in `"deadbeef deadb00f"`, I would intuitively expect the result to be a single nulldata output with multiple pushed like `OP_RETURN <deadbeef> <deadb00f>` instead of multiple OP_RETURN outputs..

Also did you check if `RPCArg::Type::STR_HEX` actually allows spaces? The doc says
```
        STR_HEX,       //!< Special type that is a STR with only hex chars
```

Reply:
There are two issues here -- one is that if you want to specify multiple hex strings you should use an array rather than spaces (which are not used anywhere else and also are likely to be confusing when calling from sh). The other is whether we should have multiple outputs or multiple pushes in the one output.

Personally I think either is fine as long as it's documented.

Reply:
> Also did you check if RPCArg::Type::STR_HEX actually allows spaces?  

The functional test exercises this, the spaces are accepted

> if you want to specify multiple hex strings you should use an array rather than spaces 

This was the breaking change previously? Or is it that you want a different separator? 

> Personally I think either is fine as long as it's documented 

I agree, I'm not fussed either way please let me know what you both prefer.

Reply:
> This was the breaking change previously? Or is it that you want a different separator?
>

Try to parse as both and do the right thing based on the given type. This is done in a few other RPCs as well (or at least used to be).

Reply:
Done